File: rpm-ws/src/main/java/ec/gob/ventanilla/entity/PubPersona.java
package ec.gob.ventanilla.entity;


import com.fasterxml.jackson.annotation.JsonIgnore;
import com.google.gson.annotations.Expose;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "pub_persona")
public class PubPersona {

    @Id
    @Column(name = "id", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String cedRuc;
    private String nombres;
    private String apellidos;
    private String tipoDocumento;
    @Temporal(TemporalType.TIMESTAMP)
    @JsonIgnore
    private Date fechaCreacion;
    @Expose(deserialize = false, serialize = false)
    @JsonIgnore
    private Date fechaNacimiento;
    private String direccion;
    private String telefono1;
    private String telefono2;
    private String correo1;
    private Boolean estado = true;
    @JsonIgnore
    @OneToOne(mappedBy = "persona")
    private AclUser usuario;
    private String estadoCivil;
    private String nombreConyuge;
    private String apellidoConyuge;
    private String fechaExpiracion;
    @Column(name = "fecha_expedicion")
    @Expose(deserialize = false, serialize = false)
    @Temporal(TemporalType.DATE)
    @JsonIgnore
    private Date fechaExpedicion;
    @Transient
    private String correoCodificado;
    @Transient
    private Long fechaNacimientoLong;
    //CAMPO PARA LA EXPEDICION DE LA CEDULA E INICIO DE ACTIVIDADES
    @Transient
    private Long fechaExpedicionLong;

    public PubPersona() {
    }


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getCedRuc() {
        return cedRuc;
    }

    public void setCedRuc(String cedRuc) {
        this.cedRuc = cedRuc;
    }

    public String getNombres() {
        return nombres;
    }

    public void setNombres(String nombres) {
        this.nombres = nombres;
    }

    public String getTipoDocumento() {
        return tipoDocumento;
    }

    public void setTipoDocumento(String tipoDocumento) {
        this.tipoDocumento = tipoDocumento;
    }

    public Date getFechaCreacion() {
        return fechaCreacion;
    }

    public void setFechaCreacion(Date fechaCreacion) {
        this.fechaCreacion = fechaCreacion;
    }

    public Date getFechaNacimiento() {
        return fechaNacimiento;
    }

    public void setFechaNacimiento(Date fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }

    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public String getTelefono1() {
        return telefono1;
    }

    public void setTelefono1(String telefono1) {
        this.telefono1 = telefono1;
    }

    public String getTelefono2() {
        return telefono2;
    }

    public void setTelefono2(String telefono2) {
        this.telefono2 = telefono2;
    }

    public String getCorreo1() {
        return correo1;
    }

    public void setCorreo1(String correo1) {
        this.correo1 = correo1;
    }

    public Boolean getEstado() {
        return estado;
    }

    public void setEstado(Boolean estado) {
        this.estado = estado;
    }

    public AclUser getUsuario() {
        return usuario;
    }

    public void setUsuario(AclUser usuario) {
        this.usuario = usuario;
    }

    public String getApellidos() {
        return apellidos;
    }

    public void setApellidos(String apellidos) {
        this.apellidos = apellidos;
    }

    public String getEstadoCivil() {
        return estadoCivil;
    }

    public void setEstadoCivil(String estadoCivil) {
        this.estadoCivil = estadoCivil;
    }


    @Override
    public int hashCode() {
        final int prime = 31;
        int result = 1;
        result = prime * result + ((id == null) ? 0 : id.hashCode());
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        PubPersona other = (PubPersona) obj;
        if (id == null) {
            if (other.id != null) {
                return false;
            }
        } else if (!id.equals(other.id)) {
            return false;
        }
        return true;
    }

    public String getNombreConyuge() {
        return nombreConyuge;
    }

    public void setNombreConyuge(String nombreConyuge) {
        this.nombreConyuge = nombreConyuge;
    }

    public String getApellidoConyuge() {
        return apellidoConyuge;
    }

    public void setApellidoConyuge(String apellidoConyuge) {
        this.apellidoConyuge = apellidoConyuge;
    }


    public Long getFechaNacimientoLong() {
        return fechaNacimientoLong;
    }

    public void setFechaNacimientoLong(Long fechaNacimientoLong) {
        this.fechaNacimientoLong = fechaNacimientoLong;
    }

    public Long getFechaExpedicionLong() {
        return fechaExpedicionLong;
    }

    public void setFechaExpedicionLong(Long fechaExpedicionLong) {
        this.fechaExpedicionLong = fechaExpedicionLong;
    }

    public Date getFechaExpedicion() {
        return fechaExpedicion;
    }

    public void setFechaExpedicion(Date fechaExpedicion) {
        this.fechaExpedicion = fechaExpedicion;
    }

    public String getCorreoCodificado() {
        return correoCodificado;
    }

    public void setCorreoCodificado(String correoCodificado) {
        this.correoCodificado = correoCodificado;
    }

    public String getFechaExpiracion() {
        return fechaExpiracion;
    }

    public void setFechaExpiracion(String fechaExpiracion) {
        this.fechaExpiracion = fechaExpiracion;
    }


}

File: flutter_peliculas/lib/src/models/pub-persona.dart
import 'user.dart';

part 'pub-persona.g.dart';

class PubPersona {
  int? id;
  String? cedRuc;
  String? nombres;
  String? apellidos;
  String? tipoDocumento;
  String? tipoPersona;
  int? fechaCreacion;
  int? fechaNacimiento;
  int? fechaNacimientoLong;
  int? fechaExpedicionLong;
  String? direccion;

  //celular
  String? telefono1;

  //telefono
  String? telefono2;
  String? correo1;
  String? correoCodificado;
  bool? estado = true;
  User? usuario;
  String? estadoCivil;
  int? tipoPersonaWs;
  String? numeroCasa;
  String? nombreConyuge;
  String? apellidoConyuge;

  PubPersona();

  factory PubPersona.fromJson(Map<String, dynamic> json) =>
      _$PubPersonaFromJson(json);

  PubPersona fromJson(Map<String, dynamic> json) {
    return PubPersona()
      ..id = json['id'] as int? ?? null
      ..cedRuc = json['cedRuc'] as String? ?? null
      ..nombres = json['nombres'] as String? ?? null
      ..apellidos = json['apellidos'] as String? ?? null
      ..tipoDocumento = json['tipoDocumento'] as String? ?? null
      ..tipoPersona = json['tipoPersona'] as String? ?? null
      ..fechaCreacion = json['fechaCreacion'] as int? ?? null
      ..fechaNacimiento = json['fechaNacimiento'] as int? ?? null
      ..fechaNacimientoLong = json['fechaNacimientoLong'] as int? ?? null
      ..fechaExpedicionLong = json['fechaNacimientoLong'] as int? ?? null
      ..direccion = json['direccion'] as String? ?? null
      ..telefono1 = json['telefono1'] as String? ?? null
      ..telefono2 = json['telefono2'] as String? ?? null
      ..correo1 = json['correo1'] as String? ?? null
      ..correoCodificado = json['correoCodificado'] as String? ?? null
      ..estado = json['estado'] as bool? ?? null
      ..usuario = json['usuario'] == null
          ? null
          : User().fromJson(json['usuario'] as Map<String, dynamic>)
      ..estadoCivil = json['estadoCivil'] as String? ?? null
      ..tipoPersonaWs = json['tipoPersonaWs'] as int? ?? null
      ..numeroCasa = json['numeroCasa'] as String? ?? null
      ..nombreConyuge = json['nombreConyuge'] as String? ?? null
      ..apellidoConyuge = json['apellidoConyuge'] as String? ?? null;
  }

  Map<String, dynamic> toJson() => _$PubPersonaToJson(this);

  Map<String, dynamic> json(PubPersona p) => _$PubPersonaToJson(p);
}
